curriculum/challenges/english/25-front-end-development/lab-fcc-forum-leaderboard/673c91f0b934834bc4a3ecc2.md

```diff
@@ -0,0 +1,1446 @@
+    - a `td` containing the number of replies to the post. *Hint:* use `posts_count - 1`.
+    - a `td` containing the number of views of the post.
+    - a `td` containing the time passed since the last activity.
+1. You should have an async function named `fetchData`.
+1. The `fetchData` function should request data from `forumLatest` and call `showLatestPosts` passing it the response parsed as JSON.
+1. If there's an error when fetching data, the `fetchData` function should log the error to the console.
+
+# --hints--
+
+You should have a function named `timeAgo` that takes a single argument.
+
+```js
+assert.isFunction(timeAgo);
+assert.lengthOf(timeAgo, 1);
+```
+
+When the time difference between the time passed as argument and the current time is `50` minutes, `timeAgo` should return `50m ago` but found `--fcc-actual--`.
+
+```js
+const generateTime = () => {
+    const currentTime = new Date();
+    return new Date(currentTime - (1000 * 60 * 50)).toISOString()
+}
+const expected = "50m ago";
+const actual = timeAgo(generateTime());
+assert.equal(actual, expected);
+```
+
+When the time difference between the time passed as argument and the current time is `60` minutes and `10` seconds, `timeAgo` should return `1h ago` but found `--fcc-actual--`.
+
+```js
+const generateTime = () => {
+    const currentTime = new Date();
+    return new Date(currentTime - (1000 * ( 60 * 60 + 10))).toISOString()
+}
+const expected = "1h ago";
+const actual = timeAgo(generateTime());
+assert.equal(actual, expected);
+```
+
+When the time difference between the time passed as argument and the current time is `60` minutes, `timeAgo` should return `1h ago` but found `--fcc-actual--`.
+
+```js
+const generateTime = () => {
+    const currentTime = new Date();
+    return new Date(currentTime - (1000 * 60 * 60)).toISOString()
+}
+const expected = "1h ago";
+const actual = timeAgo(generateTime());
+assert.equal(actual, expected);
+```
+
+When the time difference between the time passed as argument and the current time is `15` hours, `timeAgo` should return `15h ago` but found `--fcc-actual--`.
+
+```js
+const generateTime = () => {
+    const currentTime = new Date();
+    return new Date(currentTime - (1000 * 60 * 60 * 15)).toISOString()
+}
+const expected = "15h ago";
+const actual = timeAgo(generateTime());
+assert.equal(actual, expected);
+```
+
+When the time difference between the time passed as argument and the current time is `24` hours, `timeAgo` should return `1d ago` but found `--fcc-actual--`.
+
+```js
+const generateTime = () => {
+    const currentTime = new Date();
+    return new Date(currentTime - (1000 * 60 * 60 * 24)).toISOString()
+}
+const expected = "1d ago";
+const actual = timeAgo(generateTime());
+assert.equal(actual, expected);
+```
+
+When the time difference between the time passed as argument and the current time is `25` hours, `timeAgo` should return `1d ago` but found `--fcc-actual--`.
```

Comment:
this also to check for round vs floor it would need to be like 40 hours should give 1 day.

This test would still pass doing `Math.round(25 / 24)` which gives `1`

Instead there is a difference between `Math.floor(40 / 24)` and `Math.round(40 / 24)`